File: app/src/main/java/com/lzm/lightLive/http/danmu/basic/WebSocketListener.java
package com.lzm.lightLive.http.danmu.basic;

import org.java_websocket.handshake.ServerHandshake;
import java.nio.ByteBuffer;

public abstract class WebSocketListener {

    public void onError(Exception e) {}
    public void onMessage(String message) {}
    public void onMessage(ByteBuffer byteBuffer) {}
    public void onMessage(DanMu danMu) {}
    public void onOpen(ServerHandshake handshake) {}
    public void onClose(int code, String reason, boolean remote) {}

}

File: app/src/main/java/com/lzm/lightLive/http/danmu/basic/WebSocketListener.kt
package com.lzm.lightLive.http.danmu.basic

import org.java_websocket.handshake.ServerHandshake
import java.nio.ByteBuffer

abstract class WebSocketListener {
    fun onError(e: Exception?) {}
    fun onMessage(message: String?) {}
    fun onMessage(byteBuffer: ByteBuffer?) {}
    open fun onMessage(danMu: DanMu) {}
    fun onOpen(handshake: ServerHandshake?) {}
    fun onReady() {}
    fun onClose(code: Int, reason: String?, remote: Boolean) {}
}